Add doc for a few error messages (#3764)
powsybl/powsybl-core: docs/user/frequent_errors/error_messages.md

## No NetworkFactoryService providers found

This most likely happened because you tried to read a file, but you didn't have any implementation of the interface used for reading those files.
Use the implementation provided by PowSyBl / a third party, or write your own.

> #### Fix missing impl

Author: i was not sure what you meant by fix, I assumed you meant the heading level, in which case it's done in 473796014282

# Frequent error messages

## No NetworkFactoryService providers found

> This most likely happened because you tried to read a file, but you didn't have any implementation of the interface used for reading those files.

Reviewer: ```suggestion
This most likely happened because you tried to read a file, but didn't have any implementation of the IIDM API in your classpath.
```

# Frequent error messages

If you wish to use the latest version of PowSyBl's library without having to specify versions, you should have a look at [powsybl-dependencies](https://github.com/powsybl/powsybl-dependencies)
and related examples.

### No NetworkFactoryService providers found

This most likely happened because you tried to read a file, but you didn't have any implementation of the interface used for reading those files.
Use the implementation provided by PowSyBl / a third party, or write your own.

#### Fix missing impl

If you wish use to use PowSyBl's implementation, add the following to your `pom.xml` in the `<dependencies>` section:

```xml
<dependency>
    <groupId>com.powsybl</groupId>
    <artifactId>powsybl-iidm-impl</artifactId>
    <version>$version$</version>
</dependency>
```

> Please replace `$version$` by the version of this impl that will work with other PowSyBl packages.

Reviewer: ```suggestion
Replace `$version$` by the version of this impl that will work with other PowSyBl packages.
```

A comprehensive table of matching versions can be found at [powsybl-dependencies](https://github.com/powsybl/powsybl-dependencies?tab=readme-ov-file#powsybl-included-repositories-versions);
use the `powsybl-core` version that matches for you.

You should also make sure that you have an implementation of `SerDe` if you are trying to read a file. See [adding SerDe](#fix-missing-serde).

### Unsupported file format or invalid file

You are most likely missing an implementation of `SerDe`. Use the implementation provided by PowSyBl / a third party, or write your own.

> #### Fix missing SerDe

Reviewer: ```suggestion
#### Example for an IIDM file
```

> If you wish use to use PowSyBl's implementation, add the following to your `pom.xml` in the `<dependencies>` section:

Reviewer: ```suggestion
If you wish to import an IIDM file using powsybl-core implementation, add the following to your `pom.xml` in the `<dependencies>` section:
```
